1.19

12rcu opened the pull request:
# 1.19 Bump

This is my first time doing anything with bukkit so a code review would be appreciated.

## Docker

I added a development setup for docker with docker-compose & modified the DOCKERFILE to serve paper 1.19

## Checkpoints

The method `getInventory()` on furnaces was removed and I couldn't find a good workaround so I modified the checkpoints as follows:
 - the name of the furnace is irrelevant
 - the sum of all items within the furnace will determine what cp the player has triggered  

see [https://github.com/PaperMC/Paper/issues/2040#issuecomment-492683567](https://github.com/PaperMC/Paper/issues/2040#issuecomment-492683567)

12rcu committed:
added docker-compose.yml for dev server

12rcu committed:
load worlds

12rcu committed:
lift to 1.19

12rcu committed:
fix map vote

12rcu committed:
modify cp system -> relevant how many items are in the furnace

12rcu committed:
bump version

freggy commented on src/main/kotlin/cloud/luxor/lbwl/flash/MapVoting.kt:
```diff
@@ -70,22 +85,27 @@ class MapVoting(private val maps: List<MapConfig>) : Listener {
-        this.vote(event.whoClicked as Player, name)
+
+        //ugly af but I really don't know how to access the displayName as a String
+        val mapName = event.currentItem?.itemMeta?.displayName()
```
Maybe try to cast it into a `TextComponent` and then get the plain string by calling `TextComponent#content()`

12rcu replied:
that solved the problem see [https://github.com/Freggy/lbwl-flash/pull/5/commits/7b0fbea056ff503db4ee7e64c5379881ff1eb863](https://github.com/Freggy/lbwl-flash/pull/5/commits/7b0fbea056ff503db4ee7e64c5379881ff1eb863)

freggy commented on src/main/kotlin/cloud/luxor/lbwl/flash/Util.kt:
```diff
@@ -36,17 +37,18 @@ fun createTwitterLink(click: String, tweet: String, vararg hashtags: String): Ar
-        .event(HoverEvent(HoverEvent.Action.SHOW_TEXT, TextComponent.fromLegacyText("§7Öffnet ein Fenster zu Twitter.")));
-    return builder.create();
+    return Component.text(" $click").clickEvent(
+        net.kyori.adventure.text.event.ClickEvent.clickEvent(net.kyori.adventure.text.event.ClickEvent.Action.OPEN_URL, url.toString().replace(" ", "%20"))
```
Why are you not importing these classes and instead use their FQCN?

12rcu replied:
see [https://github.com/Freggy/lbwl-flash/pull/5/commits/c2c4cdf7a4242766f88c85d81114b3996b0d3361](https://github.com/Freggy/lbwl-flash/pull/5/commits/c2c4cdf7a4242766f88c85d81114b3996b0d3361)

freggy commented on src/main/kotlin/cloud/luxor/lbwl/flash/listener/PlayerListener.kt:
```diff
@@ -37,15 +41,17 @@ class PlayerListener(private val plugin: FlashPlugin) : Listener {
         if (state !is Furnace) return
 
-        val number = state.inventory.name!!.toInt()
+        val number = state.inventory.contents.sumOf {
+            it?.amount ?: 0
+        }
```
Do maps have to be changed for this to work?

12rcu replied:
yes

freggy reviewed (changes requested):
Thanks for the small patch :) Please consider the few comments I've made.

12rcu committed:
added imports

12rcu committed:
cast item display name to TextComponent

12rcu committed:
refactor dev setup

12rcu committed:
added sample worlds

12rcu committed:
modify respawn item to red dye

12rcu committed:
disable firework damage

12rcu commented:
I added 2 sample maps [https://github.com/Freggy/lbwl-flash/pull/5/commits/2138bce7bf9a21e5c8abf53a0d9c5cb45e4d02b8](https://github.com/Freggy/lbwl-flash/pull/5/commits/2138bce7bf9a21e5c8abf53a0d9c5cb45e4d02b8), disabled firework damage [https://github.com/Freggy/lbwl-flash/pull/5/commits/6778984d6a7ffd8fc04993dfd6616e3efc024856](https://github.com/Freggy/lbwl-flash/pull/5/commits/6778984d6a7ffd8fc04993dfd6616e3efc024856) and modified the reset item [https://github.com/Freggy/lbwl-flash/pull/5/commits/f22d717d7185b932970deea7ffe75d35b8e41b03](https://github.com/Freggy/lbwl-flash/pull/5/commits/f22d717d7185b932970deea7ffe75d35b8e41b03)

Freggy commented:
Thanks, merged it
